### PR TITLE
Added Explorer icons for CloudFormation and Lambda

### PR DESCRIPTION
**Pull request by awschristou**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Infrastructure (change which improves the lifecycle management (CI/CD, build, package, deploy, lint, etc) of the application, but does not change functionality.)
- [x] Technical debt (change which improves the maintainability of the codebase, but does not change functionality)
- [ ] Testing (change which modifies or adds test coverage, but does not change functionality)
- [ ] Documentation (change which modifies or adds documentation, but does not change functionality)

## Description

I added our cloud formation, lambda, and "collection of things" svgs and wired them up. At the same time I wrapped the vs code api `context.asAbsolutePath` with a stub function. 

In doing this, some code that was removed in #220 was pulled back in

A few other lines were adjusted for formatting and linting in files I touched.

## Testing

* Ran tests
* Ran Toolkit, looked at Explorer nodes for Lambda and CloudFormation
* Cycled between light and dark color schemes

## Screenshots (if appropriate)

image

## Checklist

- [x] I have read the **README** document
- [x] I have read the **CONTRIBUTING** document
- [x] My code follows the code style of this project
- [x] I have added tests to cover my changes
- [x] All new and existing tests passed
- [ ] A short description of the change has been added to the **CHANGELOG**

## License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/lambda/explorer/cloudFormationNodes.ts [hunk 1]
```diff
@@ -68,7 +73,7 @@ export class DefaultCloudFormationNode extends AWSTreeErrorHandlerNode implement
             this.stackNodes,
             stacks.keys(),
             key => this.stackNodes.get(key)!.update(stacks.get(key)!),
-            key => new DefaultCloudFormationStackNode(this, stacks.get(key)!)
+            key => new DefaultCloudFormationStackNode(this, stacks.get(key)!, this.getExtensionAbsolutePath)
```
Comment (mpiroc) [inline src/lambda/explorer/cloudFormationNodes.ts]: Be careful about passing unbound instance methods around. Either wrap it in a lambda:

```typescript
(arg1, arg2, /* etc  */) => this.getExtensionAbsolutePath(arg1, arg2, /* etc */)
```

or bind it:

```typescript
this.getExtensionAbsolutePath.bind(this)
```

File: src/lambda/explorer/cloudFormationNodes.ts [hunk 2]
```diff
@@ -141,7 +154,7 @@ export class DefaultCloudFormationStackNode extends AWSTreeErrorHandlerNode impl
             this.functionNodes,
             intersection(resources, functions.keys()),
             key => this.functionNodes.get(key)!.update(functions.get(key)!),
-            key => new DefaultCloudFormationFunctionNode(this, functions.get(key)!)
+            key => new DefaultCloudFormationFunctionNode(this, functions.get(key)!, this.getExtensionAbsolutePath)
```
Comment (mpiroc) [inline src/lambda/explorer/cloudFormationNodes.ts]: Unbound instance method--see above.

File: src/lambda/explorer/defaultRegionNode.ts [hunk 3]
```diff
@@ -70,7 +75,7 @@ export class RegionNodeCollection {
             this.regionNodes,
             regionMap.keys(),
             key => this.regionNodes.get(key)!.update(regionMap.get(key)!),
-            key => new DefaultRegionNode(regionMap.get(key)!)
+            key => new DefaultRegionNode(regionMap.get(key)!, this.getExtensionAbsolutePath)
```
Comment (mpiroc) [inline src/lambda/explorer/defaultRegionNode.ts]: Unbound instance method, see above.

Comment (mpiroc) [reply src/lambda/explorer/defaultRegionNode.ts]: There are several other instances of this, please address throughout the changeset.

File: src/test/lambda/explorer/cloudFormationNodes.test.ts [hunk 4]
```diff
@@ -194,17 +184,72 @@ describe('DefaultCloudFormationStackNode', () => {
...
+        const expectedPrefix = `/${fakeIconPathPrefix}/`
+
+        assert(node.iconPath !== undefined)
+        const iconPath = node.iconPath! as {
+            light: Uri,
+            dark: Uri
+        }
+
+        assert(iconPath.light !== undefined)
+        assert(iconPath.light instanceof Uri)
```
Comment (mpiroc) [inline src/test/lambda/explorer/cloudFormationNodes.test.ts]: Out of curiosity, what's the point of the `instanceof` check?

Comment (awschristou) [reply src/test/lambda/explorer/cloudFormationNodes.test.ts]: I pulled this back from #220 

`iconPath` is defined as `iconPath?: string | Uri | { light: string | Uri; dark: string | Uri } | ThemeIcon;` so I believe this rules out the other types the property could be

**Pull request merged**